test/test_pebble.py

```diff
@@ -1359,7 +1359,7 @@ def _request(
             raise resp
         if callable(resp):
             resp = resp()
-        return resp
+        return resp  # type: ignore
```

Comment:
Do we know why tge ignore ia needed now? Can we do anything else to avoid it?

Reply:
I had done some testings before ignoring this, and I think it's because the `self.responses` is `typing.List[typing.Any]`.

If a response is an exception, it's raised. If it's a callable, it's called and the returned type is an object. If none of those cases are matched, it's returned.

So the returned result type is `object | Any`, which doesn't match the method's return type `typing.Dict[str, typing.Any]`.

Casting the final return would help, like `return typing.cast(typing.Dict[str, typing.Any], resp)`. I ignored it because I think it's simpler, since the return type is already annotated with this type.

I'm merging now with the ignore, if the type casting is preferred, I can change it.

Reply:
Ok, so probably the best thing would be to tighten up the annotation for `responses` (or not overload it as much) to avoid needing the ignore or a cast. I guess the newer pyright found this but the older one didn't for some reason.

Leaving this for another day seems fine - I don't think it's worth opening an issue or PR now.

Reply:
Yeah, it's in a test file so I don't think worth spending more time on now.

Reply:
When in doubt, do this:

```py
from typing import reveal_type

reveal_type(resp)
```

I've poked around for a couple of minutes and discovered that:
- if resp is callable, that callable always raises, and the line above is misleading
- it's trivial to set the correct type, (below) and even our lambdas are auto-detcted when because they call functions that raise uconditionally

```py
self.responses: list[Callable[[], NoReturn]|dict[str, Any]|Exception] = []
```

(code above assumes `from __future__ import annotations` and `from typing import ...`)